### PR TITLE
New recipe: Pathfinder-v0.5.0

### PR DESCRIPTION
**Pull request by dylanxyz**
Build recipe for [Pathfinder](https://github.com/servo/pathfinder), a Rust drawing library for text and vector graphics using the GPU.

### CODE REVIEW

File: P/Pathfinder/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+name = "Pathfinder"
+version = v"0.5.0"
+sources = [
+    GitSource("https://github.com/servo/pathfinder.git", "7281a607a80dacc5470d7d3ec208341b6f0b0727"),
+    DirectorySource("./bundled")
+]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir
```
Comment (giordano) [inline P/Pathfinder/build_tarballs.jl:11]:
```suggestion
```

File: P/Pathfinder/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+cd ${WORKSPACE}/srcdir/pathfinder*
+
+# by default, pathfinder will compile to a static library,
+# so we change "staticlib" to "cdylib" to compile to a dynamic library
+atomic_patch -p1 ../patches/Compile-to-dynamic-lib.patch
+
+# we need cbindgen to generate the header file
+cargo install cbindgen --force --target x86_64-unknown-linux-musl
```
Comment (giordano) [inline P/Pathfinder/build_tarballs.jl:19]: We should have the environment variable `rust_host` to indicate the host rust target
```suggestion
cargo install cbindgen --force --target "${rust_host}"
```

File: P/Pathfinder/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+fi
+
+# install the library
+install -D -m 755 "../target/${rust_target}/release/libpathfinder.${dlext}" "${libdir}/libpathfinder.${dlext}"
+
+# install the licenses
+install_license ../LICENSE-APACHE ../LICENSE-MIT
+"""
+
+platforms = supported_platforms(; experimental=true)
```
Comment (giordano) [inline P/Pathfinder/build_tarballs.jl:47]: No need to use `experimental=true` anymore :wink: 
```suggestion
platforms = supported_platforms()
```

**Review by giordano — COMMENTED** [review]
Thank you so much for your contribution, very well done recipe!  Some minor comments

File: P/Pathfinder/build_tarballs.jl (hunk 2 above)
Comment (dylanxyz) [reply P/Pathfinder/build_tarballs.jl:19]: Hi. Thanks for the comments.

Pathfinder uses another rust crate called `cbindgen` to generate the `C` header files. cbindgen is a executable that runs during the build process, so installing for the rust target and running cbindgen generates the error: "Cannot execute binary file: Exec format error". I think the problem is that we have to install cbindgen for the host instead for the rust_target. Using `x86_64-unknown-linux-musl` solved this issue for me.

Comment (giordano) [reply P/Pathfinder/build_tarballs.jl:18]: I understand, and that's why I suggested to use `rust_host`, not `rust_target` :slightly_smiling_face: 
```console
% julia --compile=min -e 'using BinaryBuilderBase; BinaryBuilderBase.runshell(Platform("aarch64", "macos"); compilers=[:rust])'
sandbox:${WORKSPACE} # echo "${rust_host}"
x86_64-unknown-linux-musl
sandbox:${WORKSPACE} # echo "${rust_target}"
aarch64-apple-darwin
```

Comment (dylanxyz) [reply P/Pathfinder/build_tarballs.jl:18]: O i see, i misread. Sorry. I was looking for a variable like `rust_target` in the BinaryBuilder docs but i could not finded. Thanks, this solves the issue.

**Comment by dylanxyz** [comment]
You're welcome!

**Pull request merged**